docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6304,6 +6328,409 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
+deterministic in the data plane which entry a packet will match.  If a client
+wishes to make the matching behavior deterministic, it must use different
+priority values for any pair of table entries that the same packet matches.
+
+The `match` and `priority` fields are used to uniquely identify an entry within
+a table. Therefore, these fields cannot be modified after the entry has been
+inserted and must be provided for `MODIFY` and `DELETE` updates. When deleting
+an entry, these key fields (along with `is_default_action`) are the only fields
+considered by the server. All other fields must be ignored, even if they have
+nonsensical values (such as an invalid action field). In the case of a *keyless*
+table (the table has an empty match key), the server must reject all attempts to
+`INSERT` a match entry and return an `INVALID_ARGUMENT` error.
+
+The number of match entries that a table *should* support is indicated in P4Info
+(`size` field of `Table` message). The guarantees provided to the P4Runtime
+client are the same as the ones described in the P4~16~ specification for the
+`size` property [@P4TableProperties]. In particular, some implementations may
+not be able to always accommodate an arbitrary set of entries up to the
+requested size, and other implementations may provide the P4Runtime client with
+more entries than requested. The P4Runtime server must return
+`RESOURCE_EXHAUSTED` when a table entry cannot be inserted because of a size
+limitation. It is recommended that, for the sake of portability, P4Runtime
+clients do not try to insert additional entries once the size indicated in
+P4Info has been reached.
+
+## GenericTable Categories {#sec-generic-table-categories}
+
+All tables fall into certain "categories" which define what APIs are
+applicable on the table and the APIs.
+
+  * Regular tables: Read, INSERT, MODIFY, DELETE work as intended. Entries in
+    a table can be inserted, modified and deleted.
+    * `INSERT` : If another entry with same match key exists, then
+      `ALREADY_EXISTS` is returned. If no union has been defined, then the
+      union from the default entry of the table is used. Some tables might
+      not have a default entry, then the union is required to be set.
+      If `default_entry` is true, then it is treated as a MODIFY. Const
+      default union rules apply. Match fields and priority should not be set.
+
+~ Figure { #fig-generic-tables-insert-flow; \
+caption: "Generic Tables : INSERT decision flow" }
+![generic-tables-insert-flow]
+~
+[generic-tables-insert-flow]: \
+build/generic-tables-insert-flow.[svg,png] \
+{ height: 7cm; page-align: forcehere }
+
+    * `MODIFY` : Union field must be set since all data fields are under a
+      union. Modifies the union spec to new union or new field values if
+      same union.
+      If `default_entry` is true, then the default entry is modified. Match
+      fields and priority should not be set. Const default union rules apply.
```

Comment:
MODIFY should mention `NOT_FOUND` is returned if the entry is not found, similar to how `DELETE` does, if that is the intent.